pyextract.py

```diff
@@ -26,6 +26,82 @@ def __init__(self):
 default_cli_parameters = DefaultCLIParameters()
 
 
+class ShellRunner:
+    @staticmethod
+    def command_run(command, password=None):
+        run_cmd = command.split(" ")
+        if password is not None and "sudo" in run_cmd:
+            run_cmd.insert(0, "echo \"%s\"|" % password + "\n")
+            run_cmd.insert(run_cmd.index("sudo") + 1, "-S")
+
+        run_cmd = " ".join(run_cmd)
+        return subprocess.run(run_cmd, stdin=sys.stdin, stdout=sys.stdout, shell=True).returncode
+
+
+class CLIParametersParser:
+    def __init__(self):
+        print('Parameter Number :', len(sys.argv))
+        print('Parameter Lists  :', str(sys.argv))
+        print('Shell Name       :', str(sys.argv[0]))
+
+        arg_parser = argparse.ArgumentParser(
+            description=
+            "Extract a file with the suffix `.tar.gz` from the source path or remote path and extract to output_path."
+        )
+        arg_parser.add_argument('-o', '--output_path',
+                                type=str,
+                                nargs='+',
+                                default=default_cli_parameters.output_path,
+                                help="extract packet output path")
+        arg_parser.add_argument('-P', '--password',
+                                type=str,
+                                nargs='?',
+                                default=default_cli_parameters.password,
+                                help="extract packet and chmod with user password")
+        arg_parser.add_argument('-s', '--source_path',
+                                type=str,
+                                nargs='+',
+                                default=default_cli_parameters.source_path,
+                                help="extract packet from source path",
+                                required=True)
+        arg_parser.add_argument('-m', '--merge_file',
+                                type=str,
+                                nargs='+',
+                                default=default_cli_parameters.merge_file,
+                                help="extract packet and merge to a new file")
+        arg_parser.add_argument(
+            '-f', '--filename',
+            type=str,
+            nargs=1,
+            help=
+            "extract packet filename, the default file suffix is .tar.gz, such as: log.tar.gz",
+            required=True)
+        arg_parser.add_argument(
+            '-p', '--purge_source_file',
+            help=
+            'purge source file if is true',
+            action='store_true',
+            default=False)
+        arg_parser.add_argument(
+            '-F', '--filter_pattern',
+            type=str,
+            default=default_cli_parameters.filter_pattern,
+            help='filter the files to be merged')
```

Comment:
@nowhszh Don't forget to update README.md for usage at the same time. Thanks in advance.